BuildLoader.py

```diff
@@ -1029,6 +1029,14 @@ def pre_build(self):
         if not os.path.exists(self._fv_dir):
             os.makedirs(self._fv_dir)
 
+        # Validate HASH_TYPE_VALUE config
+        if (HASH_TYPE_VALUE[self._board._SIGN_HASH] != self._board.SIGN_HASH_TYPE):

```

Comment:
which one do we expect user to change  _SIGN_HASH or  SIGN_HASH_TYPE ?

Reply:
Ideally user is expected to configure all three,

- _SIGN_HASH (string),  SIGN_HASH_TYPE (value to set PCD)

- IPP_HASH_LIB_SUPPORTED_MASK  based on their needs.

For ease, we have derived SIGN_HASH_TYPE and IPP_HASH_LIB_SUPPORTED_MASK from _SIGN_HASH.  This happens fine if default buildloader configs are used.
Incase if user configures only _SIGN_HASH in boardconfig.py, it would cause a mismatch.